src/when.js

```diff
@@ -0,0 +1,34 @@
+var _curry2 = require('./internal/_curry2');
+var _identity = require('./internal/_identity');
+var ifElse = require('./ifElse');
+
+
+/**
+ * Creates a function that will first pass its argument to the given predicate
+ * function. If the predicate returns true, the function will return the result
```

Comment:
I originally had `If the predicate is satisfied` but was worried that may not be clear to some people what that actually means.

Happy to change it, as I'd prefer it myself.